clients/spark/core/src/main/scala/io/treeverse/clients/BulkRemoverFactory.scala

```diff
@@ -76,8 +79,24 @@ object BulkRemoverFactory {
 
       val delObjReq = new model.DeleteObjectsRequest(bucket).withKeys(removeKeys)
       val s3Client = getS3Client(hc, bucket, region, S3NumRetries)
-      val res = s3Client.deleteObjects(delObjReq)
-      res.getDeletedObjects.asScala.map(_.getKey())
+      try {
+        val res = s3Client.deleteObjects(delObjReq)
+        res.getDeletedObjects.asScala.map(_.getKey())
+      } catch {
+        case mde: MultiObjectDeleteException => {
+          // TODO(ariels): Delete one-by-one?!
+
+          // TODO(ariels): Metric!
+          val errors = mde.getErrors();
+          println(s"deleteObjects: Partial failure: ${errors.size} errors: ${errors}")
+          errors.foreach((de) => println(s"\t${de.getKey}: [${de.getCode}] ${de.getMessage}"))
```

Comment:
A suggestion, probably not for this PR:
Return the failed deletions from this function as well, and write them to another place so we'll be able to handle them (manually?) later.